### PR TITLE
Use proper gemspecs for managing plugins

### PR DESCRIPTION
**Pull request by segiddins**
@alloy got bored today and decided to rewrite this. Let me know what you think?

This fixes issues where the plugin would try and require other code in its own gem, which would fail because the gem had never actually been activated and its `require_paths` added to `$:`.

### CODE REVIEW

**Comment by alloy** [comment]
Awesome!!! I'll read the code tomorrow.

Also tell @krausefx, they were in the process of using this code.

File: .gitignore [hunk 1]
```diff
@@ -3,7 +3,6 @@
 .bundle
 .config
 .yardoc
-Gemfile.lock
```
Comment (kylef) [inline .gitignore]: :scream:

Comment (segiddins) [inline .gitignore]: It was checked in anyways

File: lib/claide/command/plugin_manager.rb [hunk 2]
```diff
@@ -71,28 +46,27 @@ def self.specification(path)
...
-          end
-        else
-          []
-        end
+      def self.plugin_gems_for_prefix(prefix)
+        glob = "#{prefix}_plugin#{Gem.suffix_pattern}"
+        Gem::Specification.latest_specs(true).map do |spec|
+          matches = spec.matches_for_glob(glob)
+          [spec, matches] unless matches.empty?
+        end.compact
```
Comment (alloy) [inline lib/claide/command/plugin_manager.rb]: Is there any negative effects in time it takes to perform this work?

File: spec/spec_helper.rb [hunk 3]
```diff
@@ -49,6 +50,8 @@ def should_raise_help(error_message)
 #
 CLAide::Command.ansi_output = false
 
+Mocha::Configuration.prevent(:stubbing_non_existent_method)
```
Comment (alloy) [inline spec/spec_helper.rb]: :+1:

**Comment by alloy** [comment]
Code-wise this looks good to me :ok_hand: Only interested in possible slowing down of loading.

File: lib/claide/command/plugin_manager.rb (hunk 2 above)
Comment (segiddins) [inline lib/claide/command/plugin_manager.rb]: @alloy it's actually the same work that `find_latest_files` was doing

Comment (alloy) [inline lib/claide/command/plugin_manager.rb]: Ok, :shipit:

File: claide.gemspec [hunk 4]
```diff
@@ -12,8 +12,7 @@ Gem::Specification.new do |s|
 
   s.summary  = "A small command-line interface framework."
 
-  s.files = Dir["lib/**/*.rb"] + %w{ README.markdown LICENSE }
-  s.require_paths = %w{ lib }
+  s.files = `git ls-files -z`.split("\0").reject { |f| f =~ /\A(spec|examples)/i }
```
Comment (floere) [inline claide.gemspec]: Some documentation would be appreciated by a reader :)

**Comment by floere** [comment]
:+1: :+1: :up:

File: claide.gemspec (hunk 4 above)
Comment (segiddins) [inline claide.gemspec]: That's just the standard. It's what `bundle gem` would produce.

Comment (floere) [inline claide.gemspec]:
```ruby
# The following line produces what `bundle gem` would produce. 
```

Comment (floere) [inline claide.gemspec]:
```ruby
# The following line produces what `bundle gem` would produce. 
```

**Comment by segiddins** [comment]
:shipit:. Just in time for CLAide 1.0, too!

**Pull request merged**